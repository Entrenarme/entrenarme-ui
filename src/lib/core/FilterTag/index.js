// @flow
import * as React from 'react';
import styled from 'styled-components';

import { regularFont } from '../../helpers/fonts';
import colors from '../../helpers/colors';

const MainChipContainer = styled.div`
  display: inline-flex;
  align-items: center;
  vertical-align: top;
  height: 30px;
  overflow: hidden;
  cursor: pointer;

  margin-right: 10px;

  background-color: ${colors.white.default};
  border-radius: 50px;
`;

const AddActionContainer = styled.div`
  cursor: pointer;
  width: 30px;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  & svg {
    width: 0.65em !important;
  }

  &:hover {
    background-color: ${colors.primary.light};
  }

  ${MainChipContainer}:hover & {
    background-color: ${colors.primary.light};
  }
`;

const BodyContainer = styled.div`
  font-family: ${regularFont};
  font-size: 0.875rem;
  color: ${colors.text};
  padding: ${props =>
    props.options.hover
      ? props.options.leftIcon
        ? '10px 12px 10px 10px'
        : '10px 10px 10px 12px'
      : '10px 12px'}
  height: 100%;
  border-left: 1px solid ${colors.gray.light};
  border-right: 1px solid ${colors.gray.light};
  display: flex;
  align-items: center;
  border-radius: ${props =>
    props.options.hover
      ? props.options.leftIcon
        ? '0px 50px 50px 0px'
        : '50px 0px 0px 50px'
      : '50px'};

  &:hover {
    background-color: ${props =>
      props.options.hover ? '' : colors.primary.light};
  }
`;

const RemoveActionContainer = styled.div`
  cursor: pointer;
  width: 30px;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  & svg {
    width: 0.5em !important;
  }

  &:hover {
    background-color: ${colors.primary.light};
  }

  ${MainChipContainer}:hover & {
    background-color: ${colors.primary.light};
  }
`;

type Props = {
  title: React.Node,
  leftIcon?: React.Node,
  rightIcon?: React.Node,
  callback: Function,
};

const FilterTag = ({
  title,
  callback,
  leftIcon,
  rightIcon,
  ...rest
}: Props) => {
  return (
    <MainChipContainer onClick={() => callback()} {...rest}>
      {leftIcon ? <AddActionContainer>{leftIcon}</AddActionContainer> : null}
      <BodyContainer
        options={{ hover: leftIcon || rightIcon, leftIcon, rightIcon }}
      >
        {title}
      </BodyContainer>
      {rightIcon ? (
        <RemoveActionContainer>{rightIcon}</RemoveActionContainer>
      ) : null}
    </MainChipContainer>
  );
};

FilterTag.defaultProps = {};

export default FilterTag;
