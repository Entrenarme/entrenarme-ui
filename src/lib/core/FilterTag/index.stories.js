// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import FilterTag from './index';

storiesOf('FilterTag', module).add(
  'default',
  withInfo('Filter tag')(() => <FilterTag title="Test" />),
);
