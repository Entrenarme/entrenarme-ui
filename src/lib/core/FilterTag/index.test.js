import * as React from 'react';
import { render, getByText } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import FilterTag from './index';

test('renders "FilterTag" with title and callback', () => {
  const { container } = render(<FilterTag title="test" callback={() => {}} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('text should be the same', () => {
  const { container } = render(<FilterTag title="Hola" callback={() => {}} />);

  getByText(container, 'Hola');
});
