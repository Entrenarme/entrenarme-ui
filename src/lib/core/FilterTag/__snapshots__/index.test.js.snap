// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "FilterTag" with title and callback 1`] = `
<div
  class="sc-bdVaJa ekyklS"
>
  <div
    class="sc-htpNat jjwzWA"
  >
    test
  </div>
</div>
`;
